app/src/org/commcare/android/framework/EntityDetailFragment.java

```diff
@@ -31,37 +34,61 @@
     public static final String DETAIL_INDEX = "edf_detail_index";
     public static final String CHILD_REFERENCE = "edf_detail_reference";
     
-    private AndroidSessionWrapper asw;
+    private final AndroidSessionWrapper asw;
     private NodeEntityFactory factory;
     private EntityDetailAdapter adapter;
+    private EntityDetailAdapter.EntityDetailViewModifier modifier;
 
     public EntityDetailFragment() {
         super();
         this.asw = CommCareApplication._().getCurrentSessionWrapper();
     }
-    
+
+    public void setEntityDetailModifier(final EntityDetailAdapter.EntityDetailViewModifier edvm){
+        this.modifier = edvm;
+        if(adapter != null) {
+            adapter.setModifier(edvm);
+        }
+    }
+
+    public static final String MODIFIER_KEY = "modifier";
+
+    @Override
+    public void onSaveInstanceState(final Bundle outState) {
+        super.onSaveInstanceState(outState);
+        if(modifier instanceof Parcelable) {
+            outState.putParcelable(MODIFIER_KEY, (Parcelable)modifier);
+        } else {
+            throw new IllegalArgumentException(modifier + " must implement Parcelable!");
+        }
+    }
+
     /*
-     * (non-Javadoc)
-     * @see android.support.v4.app.Fragment#onCreateView(android.view.LayoutInflater, android.view.ViewGroup, android.os.Bundle)
-     */
+         * (non-Javadoc)
+         * @see android.support.v4.app.Fragment#onCreateView(android.view.LayoutInflater, android.view.ViewGroup, android.os.Bundle)
+         */

```

Comment:
Why the strange indentation?
Don't worry about it, we need to run the code format tool on the whole odk codebase.